### ISSUE TITLE
Strange flipping of ordering of autocompletions

### ISSUE THREAD

**Issue by DavisVaughan** [rstudio/rstudio#14597]
Run:

`x <- list(ab = 1, abc = 2)`

Type: `x$ab` and watch the suggested autocompletion flip from `ab` to `abc` even though `ab` is an exact match, feels weird to me and @kevinushey 

https://github.com/rstudio/rstudio/assets/19150088/a449894d-c743-4fda-aa70-cb06988cdd39

**Comment by LuisLauM** [comment]
The same is true for function arguments.

https://github.com/rstudio/rstudio/assets/8452334/755aa929-fc21-4da3-8208-c3017fe9af85

**Comment by kevinushey** [comment]
@LuisLauM unless I'm misunderstanding, I think you're describing a different issue? Note that RStudio defaults to showing parameters in the same order as they're declared in the underlying function. In addition, if you have Copilot integration enabled, RStudio will show the completion list above the cursor position (instead of below) to avoid obscuring ghost text generated by Copilot.

**Comment by LuisLauM** [comment]
> @LuisLauM unless I'm misunderstanding, I think you're describing a different issue? Note that RStudio defaults to showing parameters in the same order as they're declared in the underlying function. In addition, if you have Copilot integration enabled, RStudio will show the completion list above the cursor position (instead of below) to avoid obscuring ghost text generated by Copilot.

Indeed. I have recently started using Copilot and I did not know that the order of the completions changes because of that. My mistake, my apology.
